src/main/scala/uk/sky/fs2/kafka/topicloader/TopicLoader.scala

```diff
@@ -77,21 +79,27 @@ trait TopicLoader {
       topics: NonEmptyList[String],
       strategy: LoadTopicStrategy,
       consumer: KafkaConsumer[F, K, V]
-  ): Stream[F, ConsumerRecord[K, V]] = Stream
-    .eval({
+  ): Stream[F, ConsumerRecord[K, V]] = Stream.eval {
+    val logger = LoggerFactory[F].getLogger
+    {
       for {
         logOffsets <- OptionT(logOffsetsForTopics(topics, strategy, consumer))
+        _          <- OptionT.liftF(logger.debug(s"Log Offsets for topics: ${logOffsets.show}"))
+        _          <- OptionT.liftF(logger.debug(s"Assigning partitions: ${logOffsets.keys.mkString_(",")}"))
         _          <- OptionT.liftF(consumer.assign(logOffsets.keys))
-        _          <- OptionT.liftF(logOffsets.toNel.traverse((tp, o) => consumer.seek(tp, o.lowest)))
+        _          <- OptionT.liftF(logOffsets.toNel.traverse { (tp, o) =>
+                        logger.debug(s"Seeking to offset ${o.lowest} for partition ${tp.show}") *>
+                          consumer.seek(tp, o.lowest)
+                      })
       } yield consumer.records.map(_.record).through(filterBelowHighestOffset(logOffsets))
-    }.getOrElse(Stream.empty))
-    .flatten
+    }.getOrElse(Stream.empty)
+  }.flatten
 
   private def filterBelowHighestOffset[F[_] : Monad : LoggerFactory, K, V](
       logOffsets: NonEmptyMap[TopicPartition, LogOffsets]
   ): Pipe[F, ConsumerRecord[K, V], ConsumerRecord[K, V]] = {
     val nonEmptyOffsets =
-      logOffsets.toSortedMap.filter((_, o) => o.highest > 0)
+      logOffsets.toSortedMap.filter((_, o) => o.highest > o.lowest)
```

Comment:
This was incorrect, as the lowest offset when the topic policy is set to delete may not be 0.